### PR TITLE
[feature/OPT-711] to [feature/OPT-629]

### PR DESCRIPTION
**Pull request by rmenciac**

### CODE REVIEW

File: tests/integration/cli/validate/test_cli_validate.py [hunk 1]
```diff
@@ -48,90 +52,166 @@ def test_validate_otm_yaml_file(self, caplog):
...
+
+    @pytest.mark.parametrize('mapping_file, mapping_file_type', [
+        pytest.param(MTMT_MAPPING_FILENAME_VALID, 'CLOUDFORMATION', id="with a valid MTMT mapping file and an invalid "
+                                                                       "CLOUDFORMATION type"),
+        pytest.param(VISIO_VALID_MAPPING_FILE, 'TERRAFORM', id="with a valid VISIO mapping file and an invalid "
+                                                               "TERRAFORM type"),
+        pytest.param(CFT_VALID_MAPPING_FILENAME, 'VISIO',
+                     id="with a valid CLOUDFORMATION mapping file and an invalid VISIO type"),
+        pytest.param(TF_VALID_MAPPING_FILE, 'MTMT', id="with a valid TERRAFORM mapping file and an invalid MTMT type")])
+    def test_validate_valid_mapping_file_and_invalid_type(self, mapping_file, mapping_file_type):
```
Comment (galbiach-irius) [inline tests/integration/cli/validate/test_cli_validate.py:196]: Should we rename this test to "test_validate_valid_mapping_file_and_wrong_type", and create a specific test for invalid types (formats that are not implemented)?

**Review by galbiach-irius — CHANGES_REQUESTED** [review]

File: startleft/startleft/cli/cli.py [hunk 2]
```diff
@@ -189,29 +190,34 @@ def parse_any(iac_type, diagram_type, etm_type, mapping_file, default_mapping_fi
...
-    if diagram_mapping_file:
-        # TODO Cannot assume all diagram/iac mapping file will have the same validation type, so we need to request the type in the command
-        logger.info("Validating Diagram mapping files")
-        provider_resolver.get_mapping_validator(DiagramType.VISIO, [get_data(diagram_mapping_file)]).validate()
-
-    if etm_mapping_file:
-        logger.info("Validating External Threat Model mapping files")
-        provider_resolver.get_mapping_validator(EtmType.MTMT, [get_data(etm_mapping_file)]).validate()
+    if mapping_file:
+        if mapping_type is not None:
```
[the remark this answers is not in the document]
Comment (rmenciac) [reply startleft/startleft/cli/cli.py:216]: Nested if has been removed

File: startleft/startleft/messages.py [hunk 3]
```diff
@@ -70,10 +70,12 @@
...
 
-IAC_MAPPING_FILE_NAME = '--iac-mapping-file'
-IAC_MAPPING_FILE_SHORTNAME = '-i'
-IAC_MAPPING_FILE_DESC = 'IaC mapping file to validate.'
+VALIDATE_MAPPING_FILE_NAME = '--mapping-file'
+VALIDATE_MAPPING_FILE_SHORTNAME = '-m'
+VALIDATE_MAPPING_FILE_DESC = 'Mapping file to validate.'
+
+VALIDATE_MAPPING_FILE_TYPE_NAME = '--mapping-type'
+VALIDATE_MAPPING_FILE_TYPE_SHORTNAME = '-y'
```
[the remark this answers is not in the document]
Comment (rmenciac) [reply startleft/startleft/messages.py:78]: The option has been changed to "-t"

File: startleft/startleft/messages.py [hunk 4]
```diff
@@ -70,10 +70,12 @@
...
+VALIDATE_MAPPING_FILE_TYPE_DESC = 'Mapping file type to validate.'
+MAPPING_FILE_TYPE_SUPPORTED = ['CLOUDFORMATION', 'TERRAFORM', 'VISIO', 'MTMT']
```
[the remark this answers is not in the document]
Comment (rmenciac) [reply startleft/startleft/messages.py:80]: LUCID logic has been included

File: tests/integration/cli/validate/test_cli_validate.py (hunk 1 above)
Comment (rmenciac) [reply tests/integration/cli/validate/test_cli_validate.py:196]: A new test for testing the wrong mapping types has been added.

**Pull request merged**